File: merchant/android/src/main/java/io/swagger/client/api/InvoiceActionsApi.java
package io.swagger.client.api;

import io.swagger.client.ApiInvoker;
import io.swagger.client.ApiException;
import io.swagger.client.Pair;

import io.swagger.client.model.*;

import java.util.*;

import com.android.volley.Response;
import com.android.volley.VolleyError;

import io.swagger.client.model.SuccessInvoice;
import io.swagger.client.model.FailInvoice;
import io.swagger.client.model.NewInvoice;
import io.swagger.client.model.ActionAnyContent;

import org.apache.http.HttpEntity;
import org.apache.http.entity.mime.MultipartEntityBuilder;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.concurrent.ExecutionException;
import java.util.concurrent.TimeoutException;

public class InvoiceActionsApi {
  String basePath = "https://pay.coinsecure.in/";
  ApiInvoker apiInvoker = ApiInvoker.getInstance();

  public void addHeader(String key, String value) {
    getInvoker().addDefaultHeader(key, value);
  }

  public ApiInvoker getInvoker() {
    return apiInvoker;
  }

  public void setBasePath(String basePath) {
    this.basePath = basePath;
  }

  public String getBasePath() {
    return basePath;
  }

  /**
  * NEW INVOICE
  * Allows user to create a new Invoice.
   * @param apikey Enter User Key
   * @param body Enter details to create an Invoice
   * @return SuccessInvoice
  */
  public SuccessInvoice createNewInvoice (String apikey, NewInvoice body) throws TimeoutException, ExecutionException, InterruptedException, ApiException {
     Object postBody = body;
  
      // verify the required parameter 'apikey' is set
      if (apikey == null) {
      VolleyError error = new VolleyError("Missing the required parameter 'apikey' when calling createNewInvoice",
      new ApiException(400, "Missing the required parameter 'apikey' when calling createNewInvoice"));
      }
  
      // verify the required parameter 'body' is set
      if (body == null) {
      VolleyError error = new VolleyError("Missing the required parameter 'body' when calling createNewInvoice",
      new ApiException(400, "Missing the required parameter 'body' when calling createNewInvoice"));
      }
  

  // create path and map variables
  String path = "/invoice/createNewInvoice".replaceAll("\\{format\\}","json");

  // query params
  List<Pair> queryParams = new ArrayList<Pair>();
      // header params
      Map<String, String> headerParams = new HashMap<String, String>();
      // form params
      Map<String, String> formParams = new HashMap<String, String>();


          headerParams.put("apikey", ApiInvoker.parameterToString(apikey));

      String[] contentTypes = {
  
      };
      String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

      if (contentType.startsWith("multipart/form-data")) {
      // file uploading
      MultipartEntityBuilder localVarBuilder = MultipartEntityBuilder.create();
  

      HttpEntity httpEntity = localVarBuilder.build();
      postBody = httpEntity;
      } else {
      // normal form params
        }

      String[] authNames = new String[] {  };

      try {
        String localVarResponse = apiInvoker.invokeAPI (basePath, path, "POST", queryParams, postBody, headerParams, formParams, contentType, authNames);
        if(localVarResponse != null){
           return (SuccessInvoice) ApiInvoker.deserialize(localVarResponse, "", SuccessInvoice.class);
        } else {
           return null;
        }
      } catch (ApiException ex) {
         throw ex;
      } catch (InterruptedException ex) {
         throw ex;
      } catch (ExecutionException ex) {
         if(ex.getCause() instanceof VolleyError) {
	    VolleyError volleyError = (VolleyError)ex.getCause();
	    if (volleyError.networkResponse != null) {
	       throw new ApiException(volleyError.networkResponse.statusCode, volleyError.getMessage());
	    }
         }
         throw ex;
      } catch (TimeoutException ex) {
         throw ex;
      }
  }

      /**
   * NEW INVOICE
   * Allows user to create a new Invoice.
   * @param apikey Enter User Key   * @param body Enter details to create an Invoice
  */
  public void createNewInvoice (String apikey, NewInvoice body, final Response.Listener<SuccessInvoice> responseListener, final Response.ErrorListener errorListener) {
    Object postBody = body;

  
    // verify the required parameter 'apikey' is set
    if (apikey == null) {
       VolleyError error = new VolleyError("Missing the required parameter 'apikey' when calling createNewInvoice",
         new ApiException(400, "Missing the required parameter 'apikey' when calling createNewInvoice"));
    }
    
    // verify the required parameter 'body' is set
    if (body == null) {
       VolleyError error = new VolleyError("Missing the required parameter 'body' when calling createNewInvoice",
         new ApiException(400, "Missing the required parameter 'body' when calling createNewInvoice"));
    }
    

    // create path and map variables
    String path = "/invoice/createNewInvoice".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    // header params
    Map<String, String> headerParams = new HashMap<String, String>();
    // form params
    Map<String, String> formParams = new HashMap<String, String>();


    headerParams.put("apikey", ApiInvoker.parameterToString(apikey));

    String[] contentTypes = {
      
    };
    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

    if (contentType.startsWith("multipart/form-data")) {
      // file uploading
      MultipartEntityBuilder localVarBuilder = MultipartEntityBuilder.create();
      

      HttpEntity httpEntity = localVarBuilder.build();
      postBody = httpEntity;
    } else {
      // normal form params
          }

      String[] authNames = new String[] {  };

    try {
      apiInvoker.invokeAPI(basePath, path, "POST", queryParams, postBody, headerParams, formParams, contentType, authNames,
        new Response.Listener<String>() {
          @Override
          public void onResponse(String localVarResponse) {
            try {
              responseListener.onResponse((SuccessInvoice) ApiInvoker.deserialize(localVarResponse,  "", SuccessInvoice.class));
            } catch (ApiException exception) {
               errorListener.onErrorResponse(new VolleyError(exception));
            }
          }
      }, new Response.ErrorListener() {
          @Override
          public void onErrorResponse(VolleyError error) {
            errorListener.onErrorResponse(error);
          }
      });
    } catch (ApiException ex) {
      errorListener.onErrorResponse(new VolleyError(ex));
    }
  }
  /**
  * 
  * 
   * @param invoiceID 
   * @return ActionAnyContent
  */
  public ActionAnyContent loadInvoice (String invoiceID) throws TimeoutException, ExecutionException, InterruptedException, ApiException {
     Object postBody = null;
  
      // verify the required parameter 'invoiceID' is set
      if (invoiceID == null) {
      VolleyError error = new VolleyError("Missing the required parameter 'invoiceID' when calling loadInvoice",
      new ApiException(400, "Missing the required parameter 'invoiceID' when calling loadInvoice"));
      }
  

  // create path and map variables
  String path = "/invoice/{invoiceID}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "invoiceID" + "\\}", apiInvoker.escapeString(invoiceID.toString()));

  // query params
  List<Pair> queryParams = new ArrayList<Pair>();
      // header params
      Map<String, String> headerParams = new HashMap<String, String>();
      // form params
      Map<String, String> formParams = new HashMap<String, String>();



      String[] contentTypes = {
  
      };
      String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

      if (contentType.startsWith("multipart/form-data")) {
      // file uploading
      MultipartEntityBuilder localVarBuilder = MultipartEntityBuilder.create();
  

      HttpEntity httpEntity = localVarBuilder.build();
      postBody = httpEntity;
      } else {
      // normal form params
        }

      String[] authNames = new String[] {  };

      try {
        String localVarResponse = apiInvoker.invokeAPI (basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType, authNames);
        if(localVarResponse != null){
           return (ActionAnyContent) ApiInvoker.deserialize(localVarResponse, "", ActionAnyContent.class);
        } else {
           return null;
        }
      } catch (ApiException ex) {
         throw ex;
      } catch (InterruptedException ex) {
         throw ex;
      } catch (ExecutionException ex) {
         if(ex.getCause() instanceof VolleyError) {
	    VolleyError volleyError = (VolleyError)ex.getCause();
	    if (volleyError.networkResponse != null) {
	       throw new ApiException(volleyError.networkResponse.statusCode, volleyError.getMessage());
	    }
         }
         throw ex;
      } catch (TimeoutException ex) {
         throw ex;
      }
  }

      /**
   * 
   * 
   * @param invoiceID 
  */
  public void loadInvoice (String invoiceID, final Response.Listener<ActionAnyContent> responseListener, final Response.ErrorListener errorListener) {
    Object postBody = null;

  
    // verify the required parameter 'invoiceID' is set
    if (invoiceID == null) {
       VolleyError error = new VolleyError("Missing the required parameter 'invoiceID' when calling loadInvoice",
         new ApiException(400, "Missing the required parameter 'invoiceID' when calling loadInvoice"));
    }
    

    // create path and map variables
    String path = "/invoice/{invoiceID}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "invoiceID" + "\\}", apiInvoker.escapeString(invoiceID.toString()));

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    // header params
    Map<String, String> headerParams = new HashMap<String, String>();
    // form params
    Map<String, String> formParams = new HashMap<String, String>();



    String[] contentTypes = {
      
    };
    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

    if (contentType.startsWith("multipart/form-data")) {
      // file uploading
      MultipartEntityBuilder localVarBuilder = MultipartEntityBuilder.create();
      

      HttpEntity httpEntity = localVarBuilder.build();
      postBody = httpEntity;
    } else {
      // normal form params
          }

      String[] authNames = new String[] {  };

    try {
      apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType, authNames,
        new Response.Listener<String>() {
          @Override
          public void onResponse(String localVarResponse) {
            try {
              responseListener.onResponse((ActionAnyContent) ApiInvoker.deserialize(localVarResponse,  "", ActionAnyContent.class));
            } catch (ApiException exception) {
               errorListener.onErrorResponse(new VolleyError(exception));
            }
          }
      }, new Response.ErrorListener() {
          @Override
          public void onErrorResponse(VolleyError error) {
            errorListener.onErrorResponse(error);
          }
      });
    } catch (ApiException ex) {
      errorListener.onErrorResponse(new VolleyError(ex));
    }
  }
}

File: merchant/javascript/src/api/InvoiceActionsApi.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/SuccessInvoice', 'model/FailInvoice', 'model/NewInvoice', 'model/ActionAnyContent'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('../model/SuccessInvoice'), require('../model/FailInvoice'), require('../model/NewInvoice'), require('../model/ActionAnyContent'));
  } else {
    // Browser globals (root is window)
    if (!root.CoinMerchantApiDocumentation) {
      root.CoinMerchantApiDocumentation = {};
    }
    root.CoinMerchantApiDocumentation.InvoiceActionsApi = factory(root.CoinMerchantApiDocumentation.ApiClient, root.CoinMerchantApiDocumentation.SuccessInvoice, root.CoinMerchantApiDocumentation.FailInvoice, root.CoinMerchantApiDocumentation.NewInvoice, root.CoinMerchantApiDocumentation.ActionAnyContent);
  }
}(this, function(ApiClient, SuccessInvoice, FailInvoice, NewInvoice, ActionAnyContent) {
  'use strict';

  /**
   * InvoiceActions service.
   * @module api/InvoiceActionsApi
   * @version 1.0B
   */

  /**
   * Constructs a new InvoiceActionsApi. 
   * @alias module:api/InvoiceActionsApi
   * @class
   * @param {module:ApiClient} apiClient Optional API client implementation to use,
   * default to {@link module:ApiClient#instance} if unspecified.
   */
  var exports = function(apiClient) {
    this.apiClient = apiClient || ApiClient.instance;


    /**
     * Callback function to receive the result of the createNewInvoice operation.
     * @callback module:api/InvoiceActionsApi~createNewInvoiceCallback
     * @param {String} error Error message, if any.
     * @param {module:model/SuccessInvoice} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * NEW INVOICE
     * Allows user to create a new Invoice.
     * @param {String} apikey Enter User Key
     * @param {module:model/NewInvoice} body Enter details to create an Invoice
     * @param {module:api/InvoiceActionsApi~createNewInvoiceCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {module:model/SuccessInvoice}
     */
    this.createNewInvoice = function(apikey, body, callback) {
      var postBody = body;

      // verify the required parameter 'apikey' is set
      if (apikey == undefined || apikey == null) {
        throw "Missing the required parameter 'apikey' when calling createNewInvoice";
      }

      // verify the required parameter 'body' is set
      if (body == undefined || body == null) {
        throw "Missing the required parameter 'body' when calling createNewInvoice";
      }


      var pathParams = {
      };
      var queryParams = {
      };
      var headerParams = {
        'apikey': apikey
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = [];
      var accepts = [];
      var returnType = SuccessInvoice;

      return this.apiClient.callApi(
        '/invoice/createNewInvoice', 'POST',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the loadInvoice operation.
     * @callback module:api/InvoiceActionsApi~loadInvoiceCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ActionAnyContent} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * @param {String} invoiceID 
     * @param {module:api/InvoiceActionsApi~loadInvoiceCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {module:model/ActionAnyContent}
     */
    this.loadInvoice = function(invoiceID, callback) {
      var postBody = null;

      // verify the required parameter 'invoiceID' is set
      if (invoiceID == undefined || invoiceID == null) {
        throw "Missing the required parameter 'invoiceID' when calling loadInvoice";
      }


      var pathParams = {
        'invoiceID': invoiceID
      };
      var queryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = [];
      var accepts = [];
      var returnType = ActionAnyContent;

      return this.apiClient.callApi(
        '/invoice/{invoiceID}', 'GET',
        pathParams, queryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }
  };

  return exports;
}));
